src/gas_price.rs

```diff
@@ -69,6 +69,18 @@ impl GasPrice1559 {
     }
 }
 
+impl std::fmt::Display for GasPrice1559 {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{{ max_fee: {:.9} gwei, max_priority_fee: {:.9} gwei, base_fee: {:.9} gwei }}",
```

Comment:
This will always print nine decimals even if you have a nice round value.
Instead I would suggest something like this to make smaller values readable as well:
```rust
        let format_unit = |wei| {
            let gwei: f64 = wei / 1e9;
            if gwei >= 1.0 {
                format!("{:.1} Gwei", gwei)
            } else {
                format!("{wei} wei")
            }
        };
        
        write!(
            f,
            "{{ max_fee: {}, max_priority_fee: {}, base_fee: {} }}",
            format_unit(self.max_fee_per_gas),
            format_unit(self.max_priority_fee_per_gas),
            format_unit(self.base_fee_per_gas),
        )
```

Reply:
I am a bit worried about misreading the values if they have different base, as on xdai, where we would have:

`current gas price is to high { max_fee: 1.9 Gwei, max_priority_fee: 1.8 Gwei, base_fee: 7 wei }`

But I guess it's fine in most cases. I would just maybe increase the number in decimals to `2`